vendor/opam/src/core/custom_opamStubs.c

```diff
@@ -78,6 +79,26 @@ CAMLprim value OPAMW_GetArchitecture(value unit)
     return get_PE_cpu_architecture(NativeMachine);
 }
 
+/*
+ * Somewhat against my better judgement, wrap SHGetFolderPath rather than
+ * SHGetKnownFolderPath to maintain XP compatibility. OPAM already requires
```

Comment:
But then, wouldn't we be perfectly fine to use `SHGetKnownFolderPath` in our stubs? We don't really have to stay compatible with an OS from 2001.

I don't even think OCaml 5+ has a native compiler that supports 32-bit systems like XP machines would predominantly be.

Reply:
I didn't write this comment, but took it from the opam stubs.

Reply:
Yes, I know, but the point of the comment is a valid one and while for OPAM it might've been the right call (I assume), it might not be for us.